rl4co/models/zoo/mdam/decoder.py

```diff
@@ -39,11 +39,18 @@ def __init__(
         train_decode_type: str = "sampling",
         val_decode_type: str = "greedy",
         test_decode_type: str = "greedy",
+        context_embedding: nn.Module = None,
+        dynamic_embedding: nn.Module = None,
+        **kwargs,
     ):
         super(Decoder, self).__init__()
-        self.dynamic_embedding = env_dynamic_embedding(
-            env_name, {"embedding_dim": embedding_dim}
-        )
+        
+        if dynamic_embedding is None:
+            self.dynamic_embedding = env_dynamic_embedding(
+                env_name, {"embedding_dim": embedding_dim}
+            )
+        else:
+            dynamic_embedding = dynamic_embedding
```

Comment:
[Minor] I think here could be `self.dynamic_embedding = dynamic_embedding`, merge with L97.

Reply:
updated thanks